gtsfm/bundle/bundle_adjustment.py

```diff
@@ -52,47 +54,44 @@ def run(self, initial_data: SfmData) -> SfmResult:
 
         # Add measurements to the factor graph
         for j in range(initial_data.number_tracks()):
-            track = initial_data.track(j)  # SfmTrack
+            track = initial_data.get_track(j)  # SfmTrack
             # retrieve the SfmMeasurement objects
             for m_idx in range(track.number_measurements()):
                 # i represents the camera index, and uv is the 2d measurement
                 i, uv = track.measurement(m_idx)
                 # note use of shorthand symbols C and P
                 graph.add(GeneralSFMFactorCal3Bundler(uv, measurement_noise, C(i), P(j)))
 
-        # Add a prior on pose x1. This indirectly specifies where the origin is.
+        # get all the valid camera indices, which need to be added to the graph.
+        valid_camera_indices = initial_data.get_valid_camera_indices()
+
+        # Add a prior on first pose. This indirectly specifies where the origin is.
         graph.push_back(
             gtsam.PriorFactorPinholeCameraCal3Bundler(
-                C(0),
-                initial_data.camera(0),
+                C(valid_camera_indices[0]),
+                initial_data.get_camera(valid_camera_indices[0]),
                 gtsam.noiseModel.Isotropic.Sigma(PINHOLE_CAM_CAL3BUNDLER_DOF, 0.1),
             )
         )
         # Also add a prior on the position of the first landmark to fix the scale
         graph.push_back(
             gtsam.PriorFactorPoint3(
-                P(0),
-                initial_data.track(0).point3(),
-                gtsam.noiseModel.Isotropic.Sigma(POINT3_DOF, 0.1),
+                P(0), initial_data.get_track(0).point3(), gtsam.noiseModel.Isotropic.Sigma(POINT3_DOF, 0.1),
             )
         )
 
         # Create initial estimate
         initial = gtsam.Values()
 
-        i = 0
         # add each PinholeCameraCal3Bundler
-        for cam_idx in range(initial_data.number_cameras()):
-            camera = initial_data.camera(cam_idx)
-            initial.insert(C(i), camera)
-            i += 1
+        for cam_idx in valid_camera_indices:
```

Comment:
i prefer we use the variable `i` here, not `cam_idx`, since `i` is a camera index.
But an inline comment in the line above saying `# i is a camera index` would be nice